extensions/2.0/Vendor/EXT_materials_bump/README.md

```diff
@@ -0,0 +1,142 @@
+# EXT\_materials\_bump
+
+## Contributors
+
+- Ben Houston, Threekit [@bhouston](https://github.com/bhouston)
+
+Copyright 2023 The Khronos Group Inc. All Rights Reserved. glTF is a trademark of The Khronos Group Inc.
+See [Appendix](#appendix-full-khronos-copyright-statement) for full Khronos Copyright Statement.
+
+## Status
+
+Draft, Ratified by the Khronos Group
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Exclusions
+
+- This extension must not be used on a material that also uses `KHR_materials_pbrSpecularGlossiness`.
+- This extension must not be used on a material that also uses `KHR_materials_unlit`.
+
+## Overview
+
+This extension adds one parameters to the metallic-roughness material: `bump`.
+
+`bump` allows users to perturb the surface normal via a bump map. The bump map is a grayscale texture, where the value of each texel represents the height of the surface at that texel.  The height is then used to perturb the normal direction.
+
+Benefits of bump maps are primarily these two:
+
+1. Bump maps are much more compact that normal maps and lead to smaller glTF files as well as less GPU memory requirements.  This is because the normal map has to store a full 3D vector per texel, while the bump map only needs to store a single scalar value per texel.  Thus bump maps require 3 times less data than normal maps for transmission and GPU storage.  Normal maps are also very sensitive to texture compressions methods, while bump maps are less sensitive.
+
+2. Bump maps are useful for capturing small details in combination with a normal map for large scale structures.  For example, when rendering human faces, there are usually two layers of surface pertubation detail, the unwrapped normal map is used for representing wrinkles, and then a repeating bump map is used for skin pore details.  This is also the case for many other materials such as wood, metal, and stone, where there is both macro and micro details.
```

Comment:
I'll just note that this "second layer for small details" is not unique to bump maps, and could also be achieved with a second normal map. Common use cases that benefit from a second layer of normals *might* also require a second layer `baseColorTexture` or `metallicRoughnessTexture` ... should that be part of the proposal?

Support for MaterialX-based subgraphs as inputs to material texture sockets would also solve these issues, though that is a more complex direction.

Reply:
I would prefer to keep this extension simple and mapping to how existing materials work in popular WebGL renders so that it is minimal effort to implement.  Note that both ThreeJS has support for [both a bump and a normal map](https://github.com/mrdoob/three.js/blob/dev/src/materials/MeshStandardMaterial.js#L34) as does BabylonJS has [support for both a bump and normal map] (https://github.com/BabylonJS/Babylon.js/blob/master/packages/dev/core/src/Materials/PBR/pbrBaseMaterial.ts#L495).

Right now glTF has the artificial limitation that it cannot represent what Three.js and Babylon.js can do.

I know that some may be purists and what to ensure that bump maps are relegated to history.  I am okay with that.  This is why I didn't suggest this as a KHR_ extension rather just an industry extension.

I am trying to get glTF to parity with what these engines can do.

(BTW Unity has support for 2 maps as well, the second is called a detail map - scroll down to the bottom: https://docs.unity3d.com/Manual/StandardShaderMaterialParameterNormalMap.html. Unity converts all bump maps to normal maps because they do not care about transmission size or GPU memory usage.)

Reply:
I don't see this extension as hurting anyone because if you export a gLTF from Babylon or ThreeJS right now, they just simply discard the bump map - so exporting is a lossy operation.  All this does is no longer discard it.

Reply:
I agree that the concept of combining different resolutions of maps is useful, and applies equally to all maps. Personally, I think the MaterialX-style procedural texture maps (which is already in progress for glTF) would be ideal for this, particularly if they can combine input images in addition to procedural generation. 

The hurt we need to be aware of is industry fragmentation. The big value of glTF is there is a "right way" to do PBR, which unifies artist workflows. We should consider carefully if we have two possible approaches that accomplish the same goal.

Reply:
> The big value of glTF is there is a "right way" to do PBR.  We should consider carefully if we have two possible approaches that accomplish the same goal.

Remember @elalish this is not a KHR_ extension, but rather just an industry pragmatic one, notice the EXT_ prefix.

I purposely was not aiming to get glTF committee blessing as I figured it wouldn't get blessing because bump maps are is viewed as the old way to do things.

Reply:
In order to get an EXT_ accepted into the Khronos registry of extension (there is no need for a vote because this isn't a KHR_ extension), one only needs to 2 implementations to adopt it.  Threekit is adopting this as we speak, thus only one other implementation is required.

Reply:
Ah, thanks, I was actually wondering what the bar was for an EXT. Fair enough.

Reply:
I will mark this as resolved if that is okay.